src/Apps/Authentication/Utils/useAuthForm.ts

```diff
@@ -21,36 +21,32 @@ export function useAuthForm({
   const { match } = useRouter()
 
   const {
-    action,
     afterSignUpAction,
     contextModule,
     copy: copyQueryParam,
     destination,
     intent,
-    kind,
     oauthLogin,
-    objectId,
-  } = match.location.query as ModalOptions
-
-  const title = computeTitle({
-    copy: copyQueryParam,
-    intent,
-    pageTitle,
-  })
+    // FIXME: Convection should link using the `afterSignUpAction` param, not `submissionId`
+    submissionId,
+  } = match.location.query as ModalOptions & { submissionId?: string }
 
+  const title = computeTitle({ copy: copyQueryParam, intent, pageTitle })
   const redirectTo = getENV("AUTHENTICATION_REDIRECT_TO")
   const signupReferer = getENV("AUTHENTICATION_REFERER")
 
-  if (action) {
-    const isClient = typeof window !== "undefined"
-
-    if (isClient) {
-      Cookies.set(
-        "afterSignUpAction",
-        JSON.stringify({ action, kind, objectId })
-      )
-    }
-  }
+  // FIXME: Convection should link using the `afterSignUpAction` param,
+  // not `submissionId` so that we don't have to do this.
+  //
+  // If there's a `submissionId` constructs an `afterSignUpAction` from it. Otherwise
+  // falls back to the `afterSignUpAction` param.
```

Comment:
I'll add a ticket for this but we should use the `afterSignUpAction` (poorly named but, one thing at a time), query param. Which will get handled by our `useAuthIntent` hook.


Would look something like:
`?afterSignUpAction%5Baction%5D=follow&afterSignUpAction%5Bkind%5D=artist&afterSignUpAction%5BobjectId%5D=andy-warhol`